server/blob/blob-export-file/src/main/java/org/apache/james/blob/export/file/LocalFileBlobExportMechanism.java

```diff
@@ -128,9 +130,12 @@ private String copyBlobToFile(BlobId blobId, Optional<FileExtension> fileExtensi
         }
     }
 
-    private String generateFileUrl(Optional<FileExtension> fileExtension) {
+    private String generateFileUrl(Optional<String> fileCustomPrefix, Optional<FileExtension> fileExtension) {
         String fileName = RandomStringUtils.random(STRING_LENGTH, WITH_LETTERS, !WITH_NUMBERS);
-        String filePathWithoutExtension = configuration.exportDirectory + "/" + fileName;
+        String fileNameWithPrefix = fileCustomPrefix
+            .map(prefix -> prefix + fileName)
```

Comment:
Do we still need such a long random generated file name after the prefix?

Reply:
we still need the random part, to guarantee the randomness, prefix here is just adding a readable string to make file names easier to understand.

Reply:
because it's long, and people suggest to remove it to get a better readability